src/Engine/Data/EnvironmentTexture.cpp

```diff
@@ -0,0 +1,739 @@
+        return true;
+    }
+};
+
+// -------------------------------------------------------------------
+using namespace Ra::Core;
+
+// -------------------------------------------------------------------
+EnvironmentTexture::EnvironmentTexture( const std::string& mapName, bool isSkybox ) :
+    m_name( mapName ),
+    m_skyData { nullptr, nullptr, nullptr, nullptr, nullptr, nullptr },
+    m_isSkyBox( isSkybox ) {
+    m_type = ( mapName.find( ';' ) != mapName.npos ) ? EnvironmentTexture::EnvMapType::ENVMAP_CUBE
+                                                     : EnvironmentTexture::EnvMapType::ENVMAP_PFM;
+    if ( m_type == EnvironmentTexture::EnvMapType::ENVMAP_PFM ) {
+        auto ext = mapName.substr( mapName.size() - 3 );
+        if ( ext != "pfm" ) { m_type = EnvironmentTexture::EnvMapType::ENVMAP_LATLON; }
+    }
+    initializeTexture();
+}
+
+void EnvironmentTexture::initializeTexture() {
+    switch ( m_type ) {
+    case EnvMapType::ENVMAP_PFM:
+        setupTexturesFromPfm();
+        break;
+    case EnvMapType::ENVMAP_CUBE:
+        setupTexturesFromCube();
+        break;
+    case EnvMapType::ENVMAP_LATLON:
+        setupTexturesFromSphericalEquiRectangular();
+        break;
+    default:
+        LOG( logERROR ) << "EnvironmentTexture::initializeTexture(): unkown EnvMapType";
+    }
+    computeSHMatrices();
+    // make the envmap cube texture
+    Ra::Engine::Data::TextureParameters params { m_name,
+                                                 GL_TEXTURE_CUBE_MAP,
+                                                 m_width,
+                                                 m_height,
+                                                 1,
+                                                 GL_RGBA,
+                                                 GL_RGBA,
+                                                 GL_FLOAT,
+                                                 GL_CLAMP_TO_EDGE,
+                                                 GL_CLAMP_TO_EDGE,
+                                                 GL_CLAMP_TO_EDGE,
+                                                 GL_LINEAR_MIPMAP_LINEAR,
+                                                 GL_LINEAR,
+                                                 (void**)( m_skyData ) };
+    m_skyTexture = std::make_unique<Ra::Engine::Data::Texture>( params );
+
+    if ( m_isSkyBox ) {
+        // make the skybox geometry
+        Aabb aabb( Vector3 { -1_ra, -1_ra, -1_ra }, Vector3 { 1_ra, 1_ra, 1_ra } );
+        Geometry::TriangleMesh skyMesh = Geometry::makeSharpBox( aabb );
+        m_displayMesh                  = std::make_unique<Ra::Engine::Data::Mesh>( "skyBox" );
+        m_displayMesh->loadGeometry( std::move( skyMesh ) );
+    }
+}
+
+void EnvironmentTexture::setupTexturesFromPfm() {
+    PfmReader::HDRIMAGE* img = PfmReader::open( m_name.c_str() );
+    m_width                  = img->width / 3;
+    m_height                 = img->height / 4;
+#pragma omp parallel for
```

Comment:
Instead, would it make sense to parallelize the two nested loops at lines 243-244 ?
We could start a parallel section here anyway to avoir re-creating the thread at each iteration.
Because with the section, openmp can scale up to 6 threads only.

Reply:
tested several "simple" solutions, noe give big improvement in computation time. As this function is executed only once at loading time, we might stay as it is now.